src/components/CytoscapeGraph/Layout/GroupCompoundLayout.ts

```diff
@@ -172,29 +176,24 @@ export default class GroupCompoundLayout {
       parent.addClass(COMPOUND_PARENT_NODE_CLASS);
       // (1.b) Set the size
       parent.style(newStyles);
-      // Save the children as jsons in the parent scratchpad for later
-      parent.scratch(CHILDREN_KEY, parent.children().jsons());
+      // Save the children in the parent scratchpad for later
+      parent.scratch(CHILDREN_KEY, parent.children());
     });
 
     //  Remove the children and its edges and add synthetic edges for every edge that touches a child node.
     let syntheticEdges = this.cy.collection();
     // Removed elements are being stored because later we will add them back.
-    const elementsToRemove = parents.children().reduce((children, child) => {
-      children.push(child);
-      return children.concat(
-        child.connectedEdges().reduce((edges, edge) => {
-          // (1.c) Create synthetic edges.
-          const syntheticEdge = this.syntheticEdgeGenerator.getEdge(edge.source(), edge.target());
-          if (syntheticEdge) {
-            syntheticEdges = syntheticEdges.add(this.cy.add(syntheticEdge));
-          }
-          edges.push(edge);
-          return edges;
-        }, [])
-      );
-    }, []);
-    // (1.d) Remove children and edges that touch a child node.
-    this.cy.remove(this.cy.collection().add(elementsToRemove));
+    children.each(child => {
+      child.connectedEdges().each(edge => {
+        // (1.c) Create synthetic edges.
+        const syntheticEdge = this.syntheticEdgeGenerator.getEdge(edge.source(), edge.target());
+        if (syntheticEdge) {
+          syntheticEdges = syntheticEdges.add(this.cy.add(syntheticEdge));
+        }
+      });
+    });
+    // (1.d) Detach all child nodes from parents.
+    children.move('null');
```

Comment:
isn't better to call `remove` ?
```suggestion
    children.remove();
```

Reply:
Why?  I would think this has minimal overhead.

Reply:
Is the `null` node document somewhere? If not, what is really happening in the background? is it creating a new node? 
I'm afraid that in the future this function would fail if the target node does not exist.

Reply:
no, this is the standard way to move to no parent:  http://js.cytoscape.org/#eles.move :

"Move the elements with respect to graph topology (i.e. new source, target, or parent)."
"parent The ID of the new parent node (use null for no parent)"

Reply:
Ah, I missed that part.